starlette/applications.py

```diff
@@ -111,9 +111,8 @@ def build_middleware_stack(self) -> ASGIApp:
     def routes(self) -> typing.List[BaseRoute]:
         return self.router.routes
 
-    # TODO: Make `__name` a positional-only argument when we drop Python 3.7 support.
-    def url_path_for(self, __name: str, **path_params: typing.Any) -> URLPath:
-        return self.router.url_path_for(__name, **path_params)
+    def url_path_for(self, /, name: str, **path_params: typing.Any) -> URLPath:
```

Comment:
Is there a mistake here?

Reply:
It isn't supposed to be... Why do you think it is?

Reply:
I recall from our earlier discussion that `name` should be a positional-only parameter.

Maybe it should be changed to `(self, name: str, /, **path_params: typing.Any)`

Reply:
Ah, it's a mistake! 🤣